node/src/components/block_executor.rs

```diff
@@ -417,9 +430,18 @@ impl BlockExecutor {
         state_root_hash: Digest,
         next_era_validator_weights: Option<BTreeMap<PublicKey, U512>>,
     ) -> Block {
-        let (parent_summary_hash, parent_seed) = if finalized_block.is_genesis_child() {
+        let (parent_summary_hash, parent_seed) = if self.is_highest_block_child(&finalized_block) {
             // Genesis, no parent summary.
-            (BlockHash::new(Digest::default()), Digest::default())
+            (
+                self.highest_block_summary
+                    .as_ref()
+                    .map(|summary| summary.hash)
+                    .unwrap_or_else(|| BlockHash::new(Digest::default())),
```

Comment:
How is using `Digest::default()` safe?

Reply:
The idea is that if we don't have the highest block summary, we're starting from genesis. So we just do what we were doing for genesis.